### ISSUE TITLE
Release

### ISSUE THREAD

**Issue by kyleabeauchamp** [choderalab/pymbar#188]
@davidlmobley has requested a beta release and conda package for pymbar3.  If there are no objections, I will roll that out ASAP.  One question is whether #186 is ready for merge, in which case I can merge that before the release.

**Comment by jchodera** [comment]
Does anyone know how to support pymbar3?

There's probably no harm in releasing a beta release, but we do need a way to handle the bugs that are found this way.  I'm still slogging through the BAR code in #186 trying to understand what it does and what the cause of instability is.

**Comment by kyleabeauchamp** [comment]
What do you mean by "does anyone know how to support pymbar3"?

**Comment by jchodera** [comment]
Sorry, that was poorly phrased: What is our model for supporting pymbar3 once it's out in the wild?  Do we have enough expertise from the Shirts and Mobley side?  Would you have time to help deal with bugs?  Or am I going to be the one it falls to in order to fix things?

I'm just trying to avoid the situation where I'm the only one tasked with supporting a beta release where I don't have a good working familiarity of the current codebase---that's all!  If we're covered, then I have no problems with cutting a 3.0.0-beta release!  If I have to support it myself, then I'd want us to wait a bit longer.

**Comment by kyleabeauchamp** [comment]
So I think Michael has some familiarity with many of the changes.  The changes he is less familiar with mostly live here: https://github.com/choderalab/pymbar/blob/master/pymbar/mbar_solvers.py.  IMHO, these changes actually bring about substantial improvements in readability.  

I think Levi has some familiarity as well.

Regarding support: I imagine I should have time for bugfixes outside of normal working hours.  That could mean minor delays on the hours to day timescale, but IMHO that's still better than most other projects.

**Comment by jchodera** [comment]
>  IMHO, these changes actually bring about substantial improvements in readability.

Agreed, this is really nice.  That should be unproblematic to support.

> Regarding support: I imagine I should have time for bugfixes outside of normal working hours.

That sounds great.  As long as we can bug you if we can't figure how to fix things, I'm totally fine with moving forward with the release.

**Comment by davidlmobley** [comment]
And just to chime in - currently your last release is a 2.1 beta release,
which I'd argue that in some sense is probably worse to support than a 3.0
beta release (and it also effectively means you have to support something
"old" as well as the new version).

(This of course is why I'm bringing it up to begin with - harder to support
two versions of an API at once).

On Wed, May 27, 2015 at 9:26 AM, John Chodera <notifications@github.com>
wrote:

> IMHO, these changes actually bring about substantial improvements in
> readability.
>
> Agreed, this is really nice. That should be unproblematic to support.
>
> Regarding support: I imagine I should have time for bugfixes outside of
> normal working hours.
>
> That sounds great. As long as we can bug you if we can't figure how to fix
> things, I'm totally fine with moving forward with the release.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/choderalab/pymbar/issues/188#issuecomment-105984847>.
>



-- 
David Mobley
dmobley@gmail.com
949-385-2436

**Comment by jchodera** [comment]
For some reason the 3.0.0beta conda linux release was building on Jenkins for 1.5 hours.  Not sure what's going on there. 

I killed it and will try again later.

**Comment by kyleabeauchamp** [comment]
I think it's running the tests, which take a bit longer than other
libraries.  If it doesn't work this time, we can disable the tests on the
recipe and update the build.
